src/components/CreateListItemFormWithReduce.js

```diff
@@ -0,0 +1,103 @@
+import { CreateItem } from "../YataClient";
+import React, { useReducer } from "react";
+import "./createListItem.css";
+
+const CONTENT_HINT = {
+  EMPTY: {
+    style: "help",
+    text: "Type in a new item between 0 and 100 characters",
+  },
+  TOO_LONG: {
+    style: "help is-danger",
+    text: "New item must be less than 100 characters",
+  },
+  VALID: {
+    style: "help is-success",
+    text: "Press enter to create a new item",
+  },
+};
+
+const DEFAULT_FORM_STATE = {
+  newItem: "",
+  hint: CONTENT_HINT.EMPTY,
+  valid: false,
+  error: null,
+};
+
+function onNewItemInputChanged(state, itemInput) {
+  const isItemValid = itemInput.length > 0 && itemInput.length <= 100;
+  let itemHint;
+  if (itemInput.length > 100) {
+    itemHint = CONTENT_HINT.TOO_LONG;
+  } else if (itemInput.length === 0) {
+    itemHint = CONTENT_HINT.EMPTY;
+  } else {
+    itemHint = CONTENT_HINT.VALID;
+  }
+  return {
+    ...state,
+    newItem: itemInput,
+    hint: itemHint,
+    valid: isItemValid,
+  };
+}
+
+function getReducer(props) {
+  return (state, action) => {
+    switch (action.type) {
+      case "inputUpdated":
+        return onNewItemInputChanged(state, action.newListContent);
+      case "submitFormSuccess":
+        return DEFAULT_FORM_STATE;
+      case "submitFormFailure":
+        return {
+          ...state,
+          error: action.error,
+        };
+      default:
+        return state;
+    }
+  };
+}
+
+function CreateListItemForm(props) {
+  const [state, dispatch] = useReducer(getReducer(props), DEFAULT_FORM_STATE);
+
+  function submitForm(event) {
```

Comment:
Reducer does not support async state updates. So, I decided to dispatch the events from this function instead once the list items are created.

The approach is taken from approach 1 of from this gist: https://gist.github.com/astoilkov/013c513e33fe95fa8846348038d8fe42